diff --git a/arch/arm/cpu/armv8/s32v234/soc.c b/arch/arm/cpu/armv8/s32v234/soc.c
index cb43790..ef76b91 100644
--- a/arch/arm/cpu/armv8/s32v234/soc.c
+++ b/arch/arm/cpu/armv8/s32v234/soc.c
@@ -469,6 +469,16 @@ int print_cpuinfo(void)
 		mxc_get_clock(MXC_ARM_CLK) / 1000000);
 	printf("Reset cause: %s\n", get_reset_cause());
 
+	int i = 0;
+	while (1) {
+		i++;
+		printf("#mn-start: %d [%s]\n", i, __func__);
+		mdelay(10);
+		printf("#mn-end\n");
+		if (i == 8)
+			break;
+	}
+
 	return 0;
 }
 #endif
