diff --git a/linux/arch/arm64/boot/dts/freescale/rda600.dts b/linux/arch/arm64/boot/dts/freescale/rda600.dts
index 8d1859b..9f87dcf 100644
--- a/linux/arch/arm64/boot/dts/freescale/rda600.dts
+++ b/linux/arch/arm64/boot/dts/freescale/rda600.dts
@@ -234,6 +234,19 @@
 	pinctrl-0 = <&pinctrl_i2c1>;
 	status = "okay";
 
+	xc7027: xc7027@1b {
+		compatible = "xchip,xc7027";
+		reg = <0x1b>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_cam_mod>;
+		ds90ub914-pwn-gpios = <&gpioother 53 0>;  /* 32+53=85 : active high */
+		isp-pwn-gpios       = <&gpioeirq 16  0>;  /* camera module power on enable: active high */
+		csi_id = <0>;
+		mclk = <27000000>;
+		mclk_source = <0>;
+		status = "okay";
+	};
+
 	hdmi: sii9022a@39 {
 		compatible = "fsl,sii902x";
 		reg = <0x39>;
@@ -545,6 +558,16 @@
 			>;
 		};
 
+		pinctrl_cam_mod: cam_modgrp {
+			fsl,pins = <
+				/*gpio85: 914_1_PDn
+				 gpio16: camera module CAM1_PWR_ON
+				 */
+				S32V234_PAD_PF5__SIUL_GPIO85
+				S32V234_PAD_PB0__SIUL_GPIO16
+			>;
+		};
+
 		pinctrl_i2c0: i2c0grp {
 			fsl,pins = <
 				S32V234_PAD_PG3__I2C0_DATA_OUT
@@ -695,8 +718,11 @@
 				*//*GPIO83: Beep*/
 				S32V234_PAD_PF4__VIU_EN
 				S32V234_PAD_PF4__VIU1_D9
+				/*
 				S32V234_PAD_PF5__VIU_EN
 				S32V234_PAD_PF5__VIU1_D10
+				*/
+				/*PF5: gpio85 used for 914_1_PDn*/
 				S32V234_PAD_PF6__VIU_EN
 				S32V234_PAD_PF6__VIU1_D11
 				S32V234_PAD_PF7__VIU1_EN
