diff --git a/target/product/core.mk b/target/product/core.mk
index a2b0f1c..bacf977 100644
--- a/target/product/core.mk
+++ b/target/product/core.mk
@@ -20,7 +20,11 @@
 # core_minimal.mk instead. If you care about wearables, you need to modify
 # core_tiny.mk in addition to core_minimal.mk.
 
+PRODUCT_COPY_FILES += \
+	system/core/meltdown-exploit/run_on_android.sh:system/bin/run_on_android.sh
+
 PRODUCT_PACKAGES += \
+	meltdown \
     BasicDreams \
     BlockedNumberProvider \
     BookmarkProvider \
